src/lib.ml

```diff
@@ -1121,15 +1135,28 @@ end = struct
         Table.add_exn db.table name x;
         x
       | _ ->
-        instantiate db name info ~stack ~hidden:(Some hidden)
+        instantiate db name info ~stack ~hidden:(`Hidden hidden)
+    end
+    | User_hidden (info, hidden) -> begin
+      match
+        match db.parent with
+        | None    -> St_not_found
+        | Some db -> find_internal db name ~stack
+      with
+      | St_found _ as x ->
+        Table.add_exn db.table name x;
+        x
```

Comment:
I was a bit surprised by this code. It's not entirely clear what we should do here and I guess in practice we'll never reach this point given that findlib has no parent.